core/services/relay/evm/evm.go

```diff
@@ -140,7 +135,26 @@ func (r *Relayer) NewMercuryProvider(rargs relaytypes.RelayArgs, pargs relaytype
 	if err != nil {
 		return nil, err
 	}
-	transmitter := mercury.NewTransmitter(r.lggr, configWatcher.ContractConfigTracker(), client, privKey.PublicKey, rargs.JobID, *relayConfig.FeedID, r.db, r.pgCfg)
+
+	// FIXME: We actually know the version here since it's in the feed ID, can
+	// we use generics to avoid passing three of this?
+	// https://smartcontract-it.atlassian.net/browse/MERC-1414
+	reportCodecV1 := reportcodecv1.NewReportCodec(*relayConfig.FeedID, r.lggr.Named("ReportCodecV1"))
```

Comment:
why are we always creating 3 if we are going to use only one? i.e move this into the switch

Reply:
It gets passed into `NewMercuryProvider` beneath. I tried to refactor this so it didn't need it but it's more complicated than you might think, hence leaving the FIXME here. This will be addressed in MERC-1414